cf-agent/verify_files.c

```diff
@@ -639,29 +607,23 @@ static PromiseResult RenderTemplateMustache(EvalContext *ctx, const Promise *pp,
     {
         unsigned char rendered_output_digest[EVP_MAX_MD_SIZE + 1] = { 0 };
         HashString(BufferData(output_buffer), BufferSize(output_buffer), rendered_output_digest, CF_DEFAULT_DIGEST);
+
         if (!HashesMatch(existing_output_digest, rendered_output_digest, CF_DEFAULT_DIGEST))
         {
-            if (a.transaction.action == cfa_warn || DONTDO)
-            {
-                cfPS(ctx, LOG_LEVEL_WARNING, PROMISE_RESULT_WARN, pp, &a,
-                     "Need to update rendering of '%s' from mustache template '%s' but policy is dry-run",
-                     pp->promiser, message);
-                result = PromiseResultUpdate(result, PROMISE_RESULT_WARN);
+            char *d = (char *) BufferData(output_buffer), *s, *last;
+
+            while ((s = strtok_r(d, "\n", &last)) != NULL) {
```

Comment:
Shouldn't this split the rendered result into lines and populate the `edcontext->file_start` list with all the lines? Why is it just appending the last line there?

Reply:
Well, originally I tried to simply append the complete output buffer. But as `SaveItemListAsFile` appends line endings to each item, it will result in an additional "\n" at the end of the edited file. That's why I decided to try to split the output buffer by newlines and append each single line via `AppendItem` (note that it's not only the last line, but I iterate through all the lines and append them as item). Furthermore, this should also have the side effect of correctly honoring `NewLineMode`

Reply:
> note that it's not only the last line, but I iterate through all the lines and append them as item

Wow, I don't know how I managed to overlook the `while`, sorry for the noise. Maybe I got confused by the name of the variable `last`.